### PR TITLE
Try to reply to invalid messages

### LINKED ISSUE #81: Not answering an invalid request could lock the client
https://github.com/pytroll/trollmoves/blob/164410247d7be1622fbc1ac645b421aaa336c7e4/trollmoves/server.py#L294-L296

Ideally we should respond with an error.

### PR DESCRIPTION
**Pull request by mraspaud**
Upon receiving an invalid message, the server is now replying if possible. 
Fixes #81

### CODE REVIEW

**Comment by adybbroe** [comment]
@mraspaud Where are we with this one?

**Comment by mraspaud** [comment]
waiting for reviews

File: trollmoves/server.py [hunk 1]
```diff
@@ -292,7 +292,16 @@ def _run(self):
...
                     address, _, payload = multiparts
                 except ValueError:
-                    LOGGER.warning("Invalid request, ignoring.")
+                    LOGGER.warning("Invalid request.")
+                    try:
+                        address = multiparts[0]
+                    except (TypeError, IndexError):
+                        LOGGER.warning("Address unknow, not sending an error message back.")
+                    else:
+                        message = Message('error', 'error', "Invalid message recieved")
```
Comment (pnuu) [inline trollmoves/server.py:301]: Typo, sholud be `received`

File: trollmoves/server.py [hunk 2]
```diff
@@ -292,7 +292,16 @@ def _run(self):
                 try:
                     address, _, payload = multiparts
                 except ValueError:
-                    LOGGER.warning("Invalid request, ignoring.")
+                    LOGGER.warning("Invalid request.")
+                    try:
+                        address = multiparts[0]
+                    except (TypeError, IndexError):
+                        LOGGER.warning("Address unknow, not sending an error message back.")
```
Comment (pnuu) [inline trollmoves/server.py:299]: Typo, should be `unknown`

File: trollmoves/server.py [hunk 3]
```diff
@@ -292,7 +292,16 @@ def _run(self):
...
+                    else:
+                        message = Message('error', 'error', "Invalid message recieved")
+                        Thread(target=self.reply_and_send,
+                               args=(self.unknown, address, message)).start()
```
Comment (pnuu) [inline trollmoves/server.py:303]: Does this thread terminate by itself?

Comment (pnuu) [reply trollmoves/server.py:303]: `self.unknown` isn't defined in the `RequestManager` class. Is it something inherited from the `Thread` class?

**Review by pnuu — COMMENTED** [review]
Couple typos and questions, otherwise LGTM.

File: trollmoves/server.py (hunk 1 above)
Comment (mraspaud) [reply trollmoves/server.py:301]:
```suggestion
                        message = Message('error', 'error', "Invalid message received")
```

File: trollmoves/server.py (hunk 2 above)
Comment (mraspaud) [reply trollmoves/server.py:299]:
```suggestion
                        LOGGER.warning("Address unknown, not sending an error message back.")
```

File: trollmoves/server.py (hunk 3 above)
Comment (pnuu) [reply trollmoves/server.py:303]: Nevermind, `self.unknown` is already there in the current `main` branch. But why does this show here in the diff as an addition?

**Review by pnuu — APPROVED** [review]

File: trollmoves/server.py (hunk 3 above)
Comment (mraspaud) [reply trollmoves/server.py:303]: yes, the thread sends the answer and dies

**Pull request merged**